extensions/domain.py

```diff
@@ -18,17 +18,116 @@
 
 from __future__ import annotations
 
+from typing import Any, Dict, List, Optional, Union
+from typing_extensions import TypedDict
+
+
+class VertexDict(TypedDict):
+    """Type for the dictionary representation of graph's vertices."""
+
+    x: float
+    y: float
+    label: str
+
+
+class EdgeDict(TypedDict):
+    """Type for the dictionary representation of graph's edges."""
+
+    src: int
+    dst: int
+    weight: int
+
+
+class GraphDict(TypedDict):
+    """Type for the dictionary representation of graph."""
+
+    vertices: List[VertexDict]
+    edges: List[EdgeDict]
+    isLabeled: bool
+    isDirected: bool
+    isWeighted: bool
```

Comment:
Why are these camel-cased and not snake cased?

Reply:
Because `isLabeled`, `isDirected`, and `isWeighted` is defined to represent the keys that are defined here:
https://github.com/oppia/oppia/blob/741cd122e84ac041ac5e52585e038ae3d86a6cdb/extensions/interactions/GraphInput/GraphInput.py#L69